### DISCUSSION TITLE
Do GetX controllers dispose themselves on Flutter Web when tab is closed?

### DISCUSSION THREAD

**Discussion by maares** [jonataslaw/getx#1934] [Q&A]
Hi guys,
Little of background, on mobile apps, you get a consistent lifecycle, whether app is on background or destroyed completely when closing it. But not entirely sure on web.

As I'm working on a personal flutter web project hosting on Firebase Hosting in development phase, I'm wondering if I will need to dispose GetX controllers my self to prevent  increase memory consumption on server side.

I have been googling, several mentions ways to identify when a page is being closed with methods such as:

```
html.window.onBeforeUnload.listen((event) {
   // code..
});
```
or

```
html.window.onUnload.listen((event) async{
  // do something
});
```

I'm able to identify all GetX controllers instances as I'm putting all controllers under tag with the window.sessionStorage, only when user is already logged in.

```
class WebSessionStorage extends GetxService {
  static WebSessionStorage get to => Get.find();
  static Storage sessionStorage = window.sessionStorage;

  late final String session;

  @override
  Future<void> onInit() async {
    super.onInit();
  }

  @override
  void onReady() {
    super.onReady();
  }

  @override
  void onClose() {
    super.onClose();
  }

  void initSessionStorage() {
    final _sessionId = getValue(USER_SESSION);
    if (_sessionId == null) {
      setValue(
        USER_SESSION,
        '${FirebaseAuthSer.to.user.value.id}-${Random().nextInt(100).toString()}',
      );
    }
    session = getValue(USER_SESSION)!;
  }

  String? getValue(String key) {
    return sessionStorage[key];
  }

  void setValue(String key, String value) {
    sessionStorage[key] = value;
  }
}
```

While using GetMiddleware to make sure I got the session created.

```
class EnsureAuthMiddleware extends GetMiddleware {
  @override
  Future<GetNavConfig?> redirectDelegate(GetNavConfig route) async {
    if (!FirebaseAuthSer.to.isLoggedInValue) {
      return GetNavConfig.fromRoute(Routes.LOGIN);
    }
    WebSessionStorage.to.initSessionStorage();
    return await super.redirectDelegate(route);
  }
}
```

In other words, do I need to worry of disposing GetX controllers, if the answer is yes, as my level of expertise is pretty much newbiie would like to hear your approches or advice on how to make a cleaner solution?

Thanks in advance

**Comment by samoray1998** [comment]
i have the same exact problem

**Comment by renanboni** [comment]
Want a piece of good advice? If you're not building an MVP consider switching to another framework, GetX will bring you way more problems than you might think :)

**Reply by codemaster-mo** [reply]
And what would your suggestion be?
